### PR TITLE
Modify rule S3776: LaYC format

### PR DESCRIPTION
**Pull request by antonioaversa**
This only migrates the structure for all projects but adds examples to the rule for C# and VB.NET only.

## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S3776/apex/rule.adoc [hunk 1]
```diff
@@ -1,14 +1,3 @@
...
-include::../message.adoc[]
-
-include::../parameters.adoc[]
-
-include::../highlighting.adoc[]
-
-endif::env-github,rspecator-view[]
+include::../why.adoc[]
+include::../resources.adoc[]
+include::../rspecator-dotnet.adoc[]
```
Comment (csaba-sagi-sonarsource) [inline rules/S3776/apex/rule.adoc:3]:
```suggestion
include::../rspecator.adoc[]
```

File: rules/S3776/csharp/complexities-example.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
+[source,csharp]
```
Comment (csaba-sagi-sonarsource) [inline rules/S3776/csharp/complexities-example.adoc:1]: You intentionally left the noncompliant/compliant diffs?

File: rules/S3776/why-dotnet.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+== Why is this an issue?
+
+https://www.sonarsource.com/docs/CognitiveComplexity.pdf[Cognitive Complexity] Complexity is a measure of how hard the control flow of a method is to understand.
+
+That is different from https://en.wikipedia.org/wiki/Computational_complexity_theory[Computational Complexity], which focuses on the resources (time and space) required for a method to run.
```
Comment (csaba-sagi-sonarsource) [inline rules/S3776/why-dotnet.adoc:5]: Optional:
Just to consider if we actually create confusion by mentioning also the computational complexity.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Left a few minor comments.

File: rules/S3776/why-dotnet.adoc (hunk 3 above)
Comment (antonioaversa) [reply rules/S3776/why-dotnet.adoc]: Removed, and made the two examples noncompliant-compliant, as explained [here](https://github.com/SonarSource/rspec/pull/2215#discussion_r1232193358).

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**